kapitan/inputs/jinja2_filters.py

```diff
@@ -51,17 +51,40 @@ def load_jinja2_filters(env):
     env.filters['shuffle'] = randomize_list
 
 
-def load_jinja2_filters_from_file(env, filter_paths):
-    for path in filter_paths:
-        try:
-            custom_filter_spec = util.spec_from_file_location(path.split('/')[-1].split('.')[0], path)
-            custom_filter_module = util.module_from_spec(custom_filter_spec)
-            custom_filter_spec.loader.exec_module(custom_filter_module)
-            for f in dir(custom_filter_module):
-                if isinstance(getattr(custom_filter_module, f), types.FunctionType):
-                    env.filters[f] = getattr(custom_filter_module, f)
-        except Exception as e:
-            raise IOError("jinja2 failed to render, could not find filter at {}: {}".format(path, e))
+def load_module_from_path(env, path):
+    """
+    loads a python module from provided path and adds it to jinja2 environment
+    filter name is same as that of function
+    """
+    try:
+        custom_filter_spec = util.spec_from_file_location(os.path.basename(path).split('.')[0],path)
+        custom_filter_module = util.module_from_spec(custom_filter_spec)
+        custom_filter_spec.loader.exec_module(custom_filter_module)
+        for function in dir(custom_filter_module):
+            if isinstance(getattr(custom_filter_module, function),
+                                    types.FunctionType):
+                logger.debug("custom filter loaded from {}".format(path))
+                env.filters[function] = getattr(custom_filter_module, function)
+    except Exception as e:
+        logger.debug("failed to find custom filter from path {}".format(path))
+        raise IOError("jinja2 failed to render, could not find filter at {}: {}".format(path, e))
+
+
+def load_jinja2_filters_from_files(env, jinja2_filter_paths):
+    """
+    iterates over all paths provided in jinja2_filter_paths list (files and directories) 
+    and calls load_module_from_path for each path
+    """
+    for path in jinja2_filter_paths:
+        if os.path.isfile(path) and path.endswith(".py"):
+            load_module_from_path(env, path)
+        elif os.path.isdir(path):
+            for filename in os.listdir(path):
+                subpath = os.path.join(path, filename)
+                if os.path.isfile(subpath) and subpath.endswith(".py"):
```

Comment:
If you restructure your code a bit, you can avoid repeating the lines:
```  if os.path.isfile(subpath) and subpath.endswith(".py"):
                    load_module_from_path(env, subpath) ``` 
and just do a check for paths before that and add all the subpaths in one list if it's a directory.  
Then for all of them iterate and load module.

Reply:
Hey @uberspot ,  I have updated the code. Moreover I also removed checking by isfile() because it returns False for files which do not exist. But I guess we want to throw error in case the name or path of file is wrong rather than failing at that stage when it would require those filters. What are  your thoughts on this ?

Reply:
If the file doesn't exist and the flag is pointing to an individual file instead of a dir, then it should fail with an exception (and you can do that in an if in the start of the function). If a dir is provided it should try and load all modules (aka anything that isfile and endswith ".py") but it shouldn't throw exceptions only if there aren't any files in the directory at all. You need an extra check in your code for that.

Reply:
What is the expected behaviour in case user passes two directories, one empty and other containing valid modules ?

Reply:
A warning for the empty dir (to warn the user about it but not stop execution) and the valid modules loaded from the other dir. At least that's what I would expect.